### PR TITLE
Add cpu and gpu examples of distil-whisper

### PR DESCRIPTION
**Pull request by yzheng124**
### 1. Why the change?

In this PR, we add cpu and gpu examples of distil-whisper using bigdl-llm's `transformer` and `optimize_model` API.

https://github.com/analytics-zoo/nano/issues/704

### CODE REVIEW

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Distil-Whisper
+
+In this directory, you will find examples on how you could apply BigDL-LLM INT4 optimizations on Distil-Whisper models. For illustration purposes, we utilize the [distil-whisper/distil-medium.en](https://huggingface.co/distil-whisper/distil-medium.en) as a reference Distil-Whisper model.
```
Comment (shane-huang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md:3]: maybe change to large as reference exmaple?

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/long-segment-recognize.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
```
Comment (shane-huang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/long-segment-recognize.py:6]: Can we merge short and long segment into the same py file? And use a parameter, e.g. '--long-seg' to control whether to use the long seg path? 
Then in README, you can merge the install & run part (remember to explain the extra --long-seq, batch size, chunk-len params). In sample output you can show 1 example output w/ long-seq param and 1 example w/o long-seq param

Comment (shane-huang) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/long-segment-recognize.py:6]: Btw, if long seg can also handle short sequences well (e.g. <30s) We can use long-seq=True as default.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+# set BigDL-Nano env variables
+source bigdl-nano-init
+
+# e.g. for a server with 48 cores per socket
+export OMP_NUM_THREADS=48
+numactl -C 0-47 -m 0 python ./recognize.py
+```
+
+#### 2.3 Sample Output
+#### 2.3.1 Short-Form Transcription
```
Comment (shane-huang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md:59]: Short Sequence

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+python ./recognize.py --repo-id-or-data-path 'hf-internal-testing/librispeech_asr_dummy' --chunk-length 0
+```
+Output:
+```log
+Inference time: xxxx s
+-------------------- Output --------------------
+[' Mr. Quilter is the apostle of the middle classes, and we are glad to welcome his gospel.']
+```
+
+#### 2.3.2 Long-Form Transcription
```
Comment (shane-huang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md:74]: Long Sequence

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+Model: [distil-whisper/distil-large-v2](https://huggingface.co/distil-whisper/distil-large-v2)
+
+Command line:
+```bash
+python ./recognize.py
```
Comment (shane-huang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md:80]: data path?

File: python/llm/example/CPU/PyTorch-Models/Model/distil-whisper/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+[' Mr. Quilter is the apostle of the middle classes, and we are glad to welcome his gospel.']
+```
+
+#### 2.4.2 Long-Form Transcription
+
+Model: [distil-whisper/distil-large-v2](https://huggingface.co/distil-whisper/distil-large-v2)
+
+Command line:
+```bash
+python ./recognize.py
```
Comment (shane-huang) [inline python/llm/example/CPU/PyTorch-Models/Model/distil-whisper/README.md:81]: data-path arguments is missing

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md (hunk 4 above)
Comment (yzheng124) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md:74]: I copied the terminology from the example in distil-whisper: 
- Short-Form Transcription: https://huggingface.co/distil-whisper/distil-large-v2#short-form-transcription 
- Long-Form Transcription: https://huggingface.co/distil-whisper/distil-large-v2#long-form-transcription. 

Do we need to align with the names in the example in distil-whisper?

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md (hunk 5 above)
Comment (yzheng124) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/distil-whisper/README.md:80]: The default arguments are for the long sequence transcription. Do I need to show arguments explicitly here?

**Review by shane-huang — APPROVED** [review]
LGTM

**Pull request merged**